JenniferLemos-tarefaS05/tarefacasas05.py

```diff
@@ -27,107 +30,129 @@ def obter_opcao():
-    print("Aluna não encontrada.")
-
-def consultar_status_aprovacao(dataset):
-    if len(dataset['alunas']) == 0:
-        print("\nNão há registros de alunas cadastradas.")
-        return
-
-    nome = input("Digite o nome da aluna: ")
-    for aluna in dataset['alunas']:
-        if aluna['nome'] == nome:
-            media = sum(aluna['notas']) / len(aluna['notas'])
-            presenca = 1 - (aluna['faltas'] / 100)  
-            if media >= 6 and presenca >= 0.8:  ## Calcula a presença / 80 porcentagem para aprovação.
-                status = "Aprovada"
-            else:
-                status = "Reprovada"
-            print(f"Aluna: {aluna['nome']}, Média: {media:.2f}, Status: {status}")
-            return
-    print("Aluna não encontrada.")
+            print("Digite um NOME e um SOBRENOME separado por espaço")
+    return nome, sobrenome
 
-main()
+def notas_aluna():
+    quantidade_notas = int(input("Qual a quantidade de notas?: "))
+    notas = []
 
+    for quantidade in range(quantidade_notas):
+        while True:
+            try:
+                nota = input(f"Insira a {quantidade + 1}ª nota:")
+                nota = float(nota.replace(',', '.'))
+                notas.append(nota)
+                break
+            except ValueError:
+                print("Entrada inválida. Por favor, insira um número válido.")
+    return notas
+
+def presenca_das_alunas():
+    quantidade_presenca = int(input("Qual a quantidade de aulas?: "))
+    presencas = []
+
+    for presente in range(quantidade_presenca):
+        while True:
+            presenca_str = input(f'Insira "True" para presente e "False" para ausente para a {presente + 1}ª aula: ')
+            if presenca_str.lower() == "true":
+                presencas.append(True)
+                break
+            elif presenca_str.lower() == "false":
+                presencas.append(False)
+                break
+            else:
+                print("Entrada inválida. Por favor utilize True para presente e False para ausente.")
+    return presencas
 
+def participacao_alunas():
+    while True:
+        participacao_str = input("Digite a nota de participação da aluna: ")
+        try:
+            participacao = float(participacao_str.replace(',', '.'))
+            break
+        except ValueError:
+            print("Entrada inválida. Digite números separados por vírgula ou ponto. Ex: 8.9 / 9,0")
+    return participacao
+
+def consultar_lista_alunas():
+    if not dataset:
+        print("Nenhuma aluna cadastrada.")
+    else:
+        print("\nLista de Alunas:")
+        for (nome, sobrenome), dados in dataset.items():
+            print(f"Nome: {nome} {sobrenome}, Turma: {dados['Turma']}")
+
+def consultar_faltas_aluna():
+    nome, sobrenome = nome_aluna()
+    if (nome, sobrenome) in dataset:
+        presencas = dataset[(nome, sobrenome)].get('Presenca')
+        if presencas is not None:
+            faltas = presenca_das_alunas.count(False)
```

Comment:
Presença das alunas é uma função. O "count" é uma função das listas, no caso de utilização, deveria ter sido utilizado na variável presencas. É importante testar todas as funções de forma individual para não deixar este tipo de detalhe passar.

**Sugestão:**
```suggestion
            faltas = presenca.count(False)
```